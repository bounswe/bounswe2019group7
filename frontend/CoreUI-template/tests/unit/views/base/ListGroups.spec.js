import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import ListGroups from '@/views/base/ListGroups'

Vue.use(BootstrapVue)

describe('ListGroups.vue', () => {
  it('has a name', () => {
    expect(ListGroups.name).toMatch('list-groups')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(ListGroups)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is ListGroups', () => {
    const wrapper = shallowMount(ListGroups)
    expect(wrapper.is(ListGroups)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(ListGroups)
    expect(wrapper.find('header.card-header > div > strong').text()).toMatch('Bootstrap list group')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(ListGroups)
    expect(wrapper.element).toMatchSnapshot()
  })
})
