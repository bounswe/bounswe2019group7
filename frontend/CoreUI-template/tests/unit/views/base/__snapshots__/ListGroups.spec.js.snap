// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroups.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               Bootstrap list group 
            </strong>
             
            <div
              class="card-header-actions"
            >
              <a
                class="card-header-action"
                href="https://bootstrap-vue.js.org/docs/components/list-group"
                rel="noreferrer noopener"
                target="_blank"
              >
                <small
                  class="text-muted"
                >
                  docs
                </small>
              </a>
            </div>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Cras justo odio
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Dapibus ac facilisis in
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Morbi leo risus
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Porta ac consectetur ac
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Vestibulum at eros
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
       
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              active items
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Cras justo odio
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              active="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Dapibus ac facilisis in
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Morbi leo risus
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Porta ac consectetur ac
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Vestibulum at eros
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
     
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              disabled items
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              disabled="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Cras justo odio
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Dapibus ac facilisis in
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Morbi leo risus
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              disabled="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Porta ac consectetur ac
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Vestibulum at eros
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
       
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              actionable items
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              event="click"
              href="#some-link"
              routertag="a"
              tag="div"
              target="_self"
            >
              Awesome link
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              active="true"
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
            >
              Link with active state
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
            >
              Action links are easy
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              disabled="true"
              event="click"
              href="#foobar"
              routertag="a"
              tag="div"
              target="_self"
            >
              Disabled link
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
     
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              buttons
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              button="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Button item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              button="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              I am a button
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              button="true"
              disabled="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              Disabled button
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              button="true"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              This is a button too
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
       
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              with badges
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              class="d-flex justify-content-between align-items-center"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              
              Cras justo odio
              
              <b-badge-stub
                event="click"
                pill="true"
                routertag="a"
                tag="span"
                target="_self"
                variant="primary"
              >
                14
              </b-badge-stub>
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              class="d-flex justify-content-between align-items-center"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              
              Dapibus ac facilisis in
              
              <b-badge-stub
                event="click"
                pill="true"
                routertag="a"
                tag="span"
                target="_self"
                variant="primary"
              >
                2
              </b-badge-stub>
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              class="d-flex justify-content-between align-items-center"
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              
              Morbi leo risus
              
              <b-badge-stub
                event="click"
                pill="true"
                routertag="a"
                tag="span"
                target="_self"
                variant="primary"
              >
                1
              </b-badge-stub>
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
     
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              variants
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
            >
              This is a default list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="primary"
            >
              This is a primary list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="secondary"
            >
              This is a secondary list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="success"
            >
              This is a success list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="danger"
            >
              This is a danger list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="warning"
            >
              This is a warning list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="info"
            >
              This is a info list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="light"
            >
              This is a light list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              routertag="a"
              tag="div"
              target="_self"
              variant="dark"
            >
              This is a dark list group item
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
       
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              variants active
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
            >
              This is a default list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="primary"
            >
              This is a primary list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="secondary"
            >
              This is a secondary list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="success"
            >
              This is a success list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="danger"
            >
              This is a danger list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="warning"
            >
              This is a warning list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="info"
            >
              This is a info list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="light"
            >
              This is a light list group item
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
              variant="dark"
            >
              This is a dark list group item
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
     
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        cols="12"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              inside cards
            </small>
          </div>
           
          <b-card-group-stub
            deck="true"
            tag="div"
          >
            <b-card-stub
              bodytag="div"
              footertag="div"
              headertag="div"
              subtitle=""
              subtitletag="h6"
              subtitletextvariant="muted"
              tag="div"
              title=""
              titletag="h4"
            >
              <div
                slot="header"
              >
                <b>
                  Card with list group
                </b>
              </div>
               
              <b-list-group-stub
                tag="div"
              >
                <b-list-group-item-stub
                  event="click"
                  href="#"
                  routertag="a"
                  tag="div"
                  target="_self"
                >
                  Cras justo odio
                </b-list-group-item-stub>
                 
                <b-list-group-item-stub
                  event="click"
                  href="#"
                  routertag="a"
                  tag="div"
                  target="_self"
                >
                  Dapibus ac facilisis in
                </b-list-group-item-stub>
                 
                <b-list-group-item-stub
                  event="click"
                  href="#"
                  routertag="a"
                  tag="div"
                  target="_self"
                >
                  Vestibulum at eros
                </b-list-group-item-stub>
              </b-list-group-stub>
               
              <p
                class="card-text mt-2"
              >
                
                Quis magna Lorem anim amet ipsum do mollit sit cillum voluptate ex
                nulla tempor. Laborum consequat non elit enim exercitation cillum aliqua
                consequat id aliqua. Esse ex consectetur mollit voluptate est in duis laboris
                ad sit ipsum anim Lorem.
              
              </p>
            </b-card-stub>
             
            <b-card-stub
              bodytag="div"
              footertag="div"
              headertag="div"
              nobody="true"
              subtitle=""
              subtitletag="h6"
              subtitletextvariant="muted"
              tag="div"
              title=""
              titletag="h4"
            >
              <div
                slot="header"
              >
                <b>
                  Card with flush list group
                </b>
              </div>
               
              <b-list-group-stub
                flush="true"
                tag="div"
              >
                <b-list-group-item-stub
                  event="click"
                  href="#"
                  routertag="a"
                  tag="div"
                  target="_self"
                >
                  Cras justo odio
                </b-list-group-item-stub>
                 
                <b-list-group-item-stub
                  event="click"
                  href="#"
                  routertag="a"
                  tag="div"
                  target="_self"
                >
                  Dapibus ac facilisis in
                </b-list-group-item-stub>
                 
                <b-list-group-item-stub
                  event="click"
                  href="#"
                  routertag="a"
                  tag="div"
                  target="_self"
                >
                  Vestibulum at eros
                </b-list-group-item-stub>
              </b-list-group-stub>
               
              <b-card-body-stub
                bodytag="div"
                subtitle=""
                subtitletag="h6"
                subtitletextvariant="muted"
                title=""
                titletag="h4"
              >
                
                Quis magna Lorem anim amet ipsum do mollit sit cillum voluptate ex
                nulla tempor. Laborum consequat non elit enim exercitation cillum aliqua
                consequat id aliqua. Esse ex consectetur mollit voluptate est in duis laboris
                ad sit ipsum anim Lorem.
              
              </b-card-body-stub>
            </b-card-stub>
          </b-card-group-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
     
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        md="6"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="footer"
          headertag="header"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
            <strong>
               List group 
            </strong>
            <small>
              custom content
            </small>
          </div>
           
          <b-list-group-stub
            tag="div"
          >
            <b-list-group-item-stub
              active="true"
              class="flex-column align-items-start"
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
            >
              <div
                class="d-flex w-100 justify-content-between"
              >
                <h5
                  class="mb-1"
                >
                  List group item heading
                </h5>
                 
                <small>
                  3 days ago
                </small>
              </div>
               
              <p
                class="mb-1"
              >
                
                Donec id elit non mi porta gravida at eget metus. Maecenas
                sed diam eget risus varius blandit.
              
              </p>
               
              <small>
                Donec id elit non mi porta.
              </small>
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              class="flex-column align-items-start"
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
            >
              <div
                class="d-flex w-100 justify-content-between"
              >
                <h5
                  class="mb-1"
                >
                  List group item heading
                </h5>
                 
                <small
                  class="text-muted"
                >
                  3 days ago
                </small>
              </div>
               
              <p
                class="mb-1"
              >
                
                Donec id elit non mi porta gravida at eget metus. Maecenas
                sed diam eget risus varius blandit.
              
              </p>
               
              <small
                class="text-muted"
              >
                Donec id elit non mi porta.
              </small>
            </b-list-group-item-stub>
             
            <b-list-group-item-stub
              class="flex-column align-items-start"
              disabled="true"
              event="click"
              href="#"
              routertag="a"
              tag="div"
              target="_self"
            >
              <div
                class="d-flex w-100 justify-content-between"
              >
                <h5
                  class="mb-1"
                >
                  Disabled List group item
                </h5>
                 
                <small
                  class="text-muted"
                >
                  3 days ago
                </small>
              </div>
               
              <p
                class="mb-1"
              >
                
                Donec id elit non mi porta gravida at eget metus. Maecenas
                sed diam eget risus varius blandit.
              
              </p>
               
              <small
                class="text-muted"
              >
                Donec id elit non mi porta.
              </small>
            </b-list-group-item-stub>
          </b-list-group-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
  </div>
</div>
`;
